components/server/src/server.rs

```diff
@@ -1070,7 +1070,13 @@ where
                 self.concurrency_manager.clone(),
             )
             .unwrap_or_else(|e| fatal!("gc worker failed to observe lock apply: {}", e));
-        if let Err(e) = gc_worker.start_auto_gc(auto_gc_config, safe_point) {
+        // table_factory of TiKvServer won't be None
```

Comment:
If you are unwrapping the `Option` below, there's no need to assert it again.

Reply:
removed the redundant check.